### DISCUSSION TITLE
How do I sort by lookup (relation) field?

### DISCUSSION THREAD

**Discussion by ammaraga** [directus/directus#13862] [Archive]
I've defined a foreign key, relation seems to be recognized by Directus, I've told Directus what to display when "id" is displayed, but when I try to sort the table, it sorts the ID, and not the underlying lookup value. Am I doing something wrong?

**Comment by rijkvanzanten** [comment]
Try using the column Directly, instead of relying on the parent M2O + the display template:
<img width="586" alt="CleanShot 2022-06-13 at 10 02 43@2x" src="https://user-images.githubusercontent.com/9141017/173371189-d5664a65-8d71-44d0-ab44-528f0eeaad73.png">

**Reply by ammaraga** [reply]
How do I get that view? I only see the IDs (both have FKs defined, but I don't see the relational table data.

![image](https://user-images.githubusercontent.com/74614063/173441600-6280bcac-a076-49ab-a3c3-466ade71d009.png)

**Reply by rijkvanzanten** [reply]
Are you on the latest version?

**Reply by ammaraga** [reply]
Pretty much (9.12.1). But I also notice that the relationship is (often) not recognized. There should be a "relationship" menu item under Schema, but there is none.

![image](https://user-images.githubusercontent.com/74614063/173442014-7a94f8bb-d533-42ea-83cf-d0d424df4a64.png)

**Reply by rijkvanzanten** [reply]
How was this field created? Directly in the database? 🤔

**Reply by ammaraga** [reply]
Yes, I've created everything in the database (tables, relationships/foreign keys, fields...)

**Reply by rijkvanzanten** [reply]
Try setting up a relational field in Directus itself and see if you missed anything when doing it manually 🙂 I'm assuming there's a foreign key constraint missing, or something along those lines 👍🏻
